### PR TITLE
Adresses issue #143, enables build with keras 2.1.2, add unit test with mobilenet

### PR DESCRIPTION
**Pull request by xadupre**

### CODE REVIEW

**Comment by wenbingl** [comment]
@xadupre , your last PR has altered dist from trusty to xenial, what's the purpose of this change?

File: onnxmltools/convert/keras/operator_converters/Conv.py [hunk 1]
```diff
@@ -4,7 +4,14 @@
 # license information.
 # --------------------------------------------------------------------------
 import numpy
-from keras.layers import Conv1D, Conv2D, Conv3D, Conv2DTranspose, Conv3DTranspose, DepthwiseConv2D
+from keras.layers import Conv1D, Conv2D, Conv3D, Conv2DTranspose, Conv3DTranspose
+
+try:
+    from keras.layers import DepthwiseConv2D
+except ImportError:
+    # keras <= 2.1.2
```
Comment (wenbingl) [inline onnxmltools/convert/keras/operator_converters/Conv.py]: According to the keras release note, DepthwiseConv2D  was introduced in Keras 2.1.5

File: onnxmltools/convert/keras/operator_converters/Activation.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 from ...common._registration import register_converter
 
 try:
-    from keras_applications.mobilenet import relu6
+    from keras.applications.mobilenet import relu6
```
Comment (wenbingl) [inline onnxmltools/convert/keras/operator_converters/Activation.py]: No version difference for this import?

File: .gitignore [hunk 3]
```diff
@@ -39,3 +39,6 @@ tests/end2end/tests/*
 tests/lightgbm/tests/*
 tests/build/*
 tests/tests/
```
Comment (wenbingl) [inline .gitignore]: should rename tests to some unusual name like _dump to avoid name misleading?

Comment (xadupre) [reply .gitignore]: Maybe, I would remove end2end testing as it mostly test keras. In a next PR...

File: .travis.yml [hunk 4]
```diff
@@ -23,11 +23,13 @@ install:
       curl -L https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o $HOME/miniconda.sh;
     fi
   - export ONNX_VERSION="";
+  - export KERAS_VERSION="";
   - if [[ "$TRAVIS_PYTHON_VERSION" == "3.5" ]]; then
       export ONNX_VERSION="==1.1.2";
     fi
   - if [[ "$TRAVIS_PYTHON_VERSION" == "3.6" ]]; then
       export ONNX_VERSION="==1.2.3";
+      export KERAS_VERSION="==2.1.2";
```
[the remark this answers is not in the document]
Comment (xadupre) [reply .travis.yml]: It was mentioned in one issue. I did go check further.

File: onnxmltools/convert/keras/operator_converters/Conv.py (hunk 1 above)
Comment (xadupre) [reply onnxmltools/convert/keras/operator_converters/Conv.py]: We should maybe reduce the number of versions we can support. The last one for sure. The number of downloads would help choosing the others I guess.

File: onnxmltools/convert/keras/operator_converters/Activation.py (hunk 2 above)
Comment (xadupre) [reply onnxmltools/convert/keras/operator_converters/Activation.py]: Not very clear from github...

File: .gitignore (hunk 3 above)
Comment (wenbingl) [reply .gitignore]: Probably not, since end2end also includes tests for CoreML.

**Comment by wenbingl** [comment]
Do you know why travis is stopping working?

File: tests/end2end/test_single_operator_with_cntk_backend.py [hunk 5]
```diff
@@ -171,6 +185,7 @@ def test_conv_4d(self):
 
         self._test_one_to_one_operator_coreml_channels_last(model, x)
 
+    @unittest.skipIf(not has_tensorflow(), reason="tensorflow is not installed")
```
Comment (wenbingl) [inline tests/end2end/test_single_operator_with_cntk_backend.py]: If every test has been added this skip, there must be something we should do differently. It's very easy to shadow the acutal test error without a notice.

Comment (wenbingl) [reply tests/end2end/test_single_operator_with_cntk_backend.py]: Given onnx and tensorflow are all not available for python 3.7, we can't claim onnxmltools has supported python 3.7, I'll remove this version from description and test configuration.

Comment (xadupre) [reply tests/end2end/test_single_operator_with_cntk_backend.py]: The fix for python 3.7 has been merged into the main branch (https://github.com/tensorflow/tensorflow/pull/21202), so every module works on Python 3.7, the binaries are not release on pypi. I suggest keeping the tests.

**Pull request closed without merging**